protocol/contracts/nexus/Relic.sol

```diff
@@ -0,0 +1,344 @@
+        uint256[] memory _amounts
+    ) external nonReentrant {
+        require(
+            msg.sender == ownerOf(_targetRelic),
+            "You don't own this Relic"
+        );
+
+        SHARDS.equipShard(msg.sender, _itemIds, _amounts);
+
+        for (uint256 i = 0; i < _itemIds.length; i++) {
+            balances[_targetRelic][_itemIds[i]] += _amounts[i];
+        }
+    }
+    // Allows users to unequip Shards from their Relic
+    function batchUnequipShard(
+        uint256 _targetRelic,
+        uint256[] memory _itemIds,
+        uint256[] memory _amounts
+    ) external nonReentrant {
+        require(
+            msg.sender == ownerOf(_targetRelic),
+            "You don't own this Relic"
+        );
+
+        SHARDS.unEquipShard(msg.sender, _itemIds, _amounts);
+
+        for (uint256 i = 0; i < _itemIds.length; i++) {
+            balances[_targetRelic][_itemIds[i]] -= _amounts[i];
+        }
+    }
+
+    // allows whitelisted contract to mint to an address
+    function mintFromContract(address _to, Enclave _selectedEnclave) external {
+        require(
+            whitelistedContracts[msg.sender],
+            "This contract is not authorised to mint"
+        );
+        uint256 tokenId = _tokenIdCounter.current();
+        enclaves[tokenId] = _selectedEnclave;
+        _tokenIdCounter.increment();
+        _safeMint(_to, tokenId);
+    }
+
+    // external temple contract whitelisting for Relic minting
+    function whitelistTemplar(address _toWhitelist) external {
+        require(msg.sender == whitelisterAddress, "Not authorised");
+        whitelisted[_toWhitelist] = true;
+    }
+    // external temple contracts giving out XP points to your Relic
+    function givePoints(uint256 _amount, uint256 _relicId) external {
+        require(whitelistedContracts[msg.sender], "Not authorised");
+        relicXP[_relicId] += _amount;
+        emit GivePoints(_relicId, _amount);
+    }
+
+    //------- Public -------//
+
+    function supportsInterface(bytes4 interfaceId)
+        public
+        view
+        override(IERC165, ERC721, ERC721Enumerable)
+        returns (bool)
+    {
+        return super.supportsInterface(interfaceId);
+    }
+
+    function onERC1155Received(
+        address,
+        address,
+        uint256,
+        uint256,
+        bytes memory
+    ) public virtual override returns (bytes4) {
+        return this.onERC1155Received.selector;
+    }
+
+    function onERC1155BatchReceived(
+        address,
+        address,
+        uint256[] memory,
+        uint256[] memory,
+        bytes memory
+    ) public virtual override returns (bytes4) {
+        return this.onERC1155BatchReceived.selector;
+    }
+
+    function onERC721Received(
+        address,
+        address,
+        uint256,
+        bytes memory
+    ) public virtual returns (bytes4) {
+        return this.onERC721Received.selector;
+    }
+
+    function getRelicXP(uint256 _relicId) public view returns (uint256) {
```

Comment:
Fine to be here, but you already have a getter implemented for free, which would just return 0 if the relicId doesn't exist

ie this is already implemented because `relicXP` is public:

`function relicXP(uint256 _relicId) external view returns (uint256);`